cfme/services/myservice/__init__.py

```diff
@@ -34,6 +35,19 @@ def __init__(self, appliance, name=None, description=None, vm_name=None):
         self.vm_name = vm_name
         self.parent = self.appliance.context
 
+    @property
+    def rest_obj(self):
+        return self.appliance.rest_api.collections.services.find_by(name=self.name)[0]
+
+    def add_resource_generic_object(self, gen_obj):
+        """ Add a generic object instance to the service """
+        with self.appliance.context.use(ViaREST):
+            self.rest_obj.action.add_resource(
+                resource=self.appliance.rest_api.collections.generic_objects.find_by(
+                    name=gen_obj.name
+                )[0]._ref_repr()
```

Comment:
any possibility of index errors here?

Reply:
Nope. The rest collection returns a list, and since we are passing a generic_object to this method, this method will only fail if the generic object doesn't exist on the appliance

Reply:
however, this should probably use `.get`, rather than `find_by`

Reply:
thanks for clarifying